### PR TITLE
plan: refine the code about deriving stats.

### PR DESCRIPTION
**Pull request by hanfei1991**
rename and assign physical plan's stats in `init` function.

### CODE REVIEW

File: plan/stats.go [hunk 1]
```diff
@@ -267,29 +268,50 @@ func (p *LogicalJoin) prepareStatsProfile() *statsProfile {
...
 		for i := p.children[0].Schema().Len(); i < p.schema.Len(); i++ {
-			p.profile.cardinality[i] = leftProfile.count
+			p.stats.cardinality[i] = leftProfile.count
 		}
 	}
-	return p.profile
+	return p.stats
+}
+
+func getSingletonStats(len int) *statsInfo {
```
Comment (shenli) [inline plan/stats.go]: Add comment for this.

File: plan/stats.go [hunk 2]
```diff
@@ -21,19 +21,19 @@ import (
 	log "github.com/sirupsen/logrus"
 )
 
-// statsProfile stores the basic information of statistics for the a plan's output. It is used for cost estimation.
-type statsProfile struct {
+// statsInfo stores the basic information of statistics for the a plan's output. It is used for cost estimation.
```
Comment (zz-jason) [inline plan/stats.go]: for a

File: plan/stats.go [hunk 3]
```diff
@@ -178,74 +179,74 @@ func getCardinality(cols []*expression.Column, schema *expression.Schema, profil
...
+	for i := range p.stats.cardinality {
+		p.stats.cardinality[i] = cardinality
 	}
 	p.inputCount = childProfile.count
-	return p.profile
+	return p.stats
 }
 
-// prepareStatsProfile prepares stats profile.
+// deriveStats prepares statsInfo profile.
```
Comment (lamxTyler) [inline plan/stats.go]: `profile` is not needed?

File: plan/stats.go [hunk 4]
```diff
@@ -43,38 +43,39 @@ func (s *statsProfile) collapse(factor float64) *statsProfile {
...
-	profile := p.profile
-	expectedCnt := p.expectedCnt
-	if expectedCnt > 0 && expectedCnt < profile.count {
-		factor := expectedCnt / profile.count
-		result := &statsProfile{count: expectedCnt}
-		for _, card := range profile.cardinality {
-			result.cardinality = append(result.cardinality, card*factor)
-		}
-		return result
+// we try to scale statsInfo to an expectCnt which must be smaller than the derived cnt.
```
Comment (lamxTyler) [inline plan/stats.go]: Capitalize the first letter.

**Pull request merged**